classes/author/DAO.inc.php

```diff
@@ -0,0 +1,171 @@
+
+class DAO extends EntityDAO
+{
+    /** @copydoc EntityDAO::$schema */
+    public $schema = PKPSchemaService::SCHEMA_AUTHOR;
+
+    /** @copydoc EntityDAO::$table */
+    public $table = 'authors';
+
+    /** @copydoc EntityDAO::$settingsTable */
+    public $settingsTable = 'author_settings';
+
+    /** @copydoc EntityDAO::$primarykeyColumn */
+    public $primaryKeyColumn = 'author_id';
+
+    /** @copydoc EntityDAO::$primaryTableColumns */
+    public $primaryTableColumns = [
+        'id' => 'author_id',
+        'email' => 'email',
+        'includeInBrowse' => 'include_in_browse',
+        'publicationId' => 'publication_id',
+        'seq' => 'seq',
+        'userGroupId' => 'user_group_id',
+    ];
+
+    /**
+     * Constructor
+     */
+    public function __construct(
+        PKPSchemaService $schemaService
+    ) {
+        parent::__construct($schemaService);
+    }
+
+    /**
+     * Instantiate a new DataObject
+     */
+    public function newDataObject(): Author
+    {
+        return App::make(Author::class);
+    }
+
+    /**
+     * @copydoc EntityDAO::get()
+     */
+    public function get(int $id): ?Author
+    {
+        return parent::get($id);
+    }
+
+    /**
+     * Get the total count of rows matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('a.' . $this->primaryKeyColumn)
+            ->pluck('a.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of publications matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->select(['a.*'])
+            ->get();
+
+        return LazyCollection::make(function () use ($rows) {
+            foreach ($rows as $row) {
+                yield $this->fromRow($row);
+            }
+        });
+    }
+
+    /**
+     * @copydoc EntityDAO::fromRow()
+     */
+    public function fromRow(stdClass $row): Author
+    {
+        $entity = parent::fromRow($row);
+
+        $locale = DB::table($this->table . ' as a')
+            ->join('publications as p', 'p.publication_id', '=', 'a.publication_id')
+            ->join('submissions as s', 'p.submission_id', '=', 's.submission_id')
+            ->where('author_id', '=', $entity->getId())
+            ->value('s.locale');
+
+        // Set the primary locale from the submission
+        $entity->setData('locale', $locale);
+
+        return $entity;
+    }
+
+    /**
+     * @copydoc EntityDAO::insert()
+     */
+    public function insert(Author $entity): int
+    {
+        $id = parent::_insert($entity);
+
+        return $id;
+    }
```

Comment:
Can be shorted to one line: `return parent::_insert($entity)`. Can we name the argument `$author` in these methods? It will just be a little clearer to people who aren't yet familiar with our use of the word "entity".

Reply:
Fixed